### DISCUSSION TITLE
Issue with reconnection to websocket or connection with more that one client to single websocket.

### DISCUSSION THREAD

**Discussion by eryk-merix** [fastapi/fastapi#6571] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
app = FastAPI(title="project")

@router.websocket("price")
async def get_price(
    websocket: WebSocket
) -> None:
    topics = ["price"]
    broadcast_consumer = BroadcastConsumer(
        websocket=websocket, group_id="price", topics_to_subscribe=topics, connection_type=PRICE_PATH
    )
    await broadcast_consumer.broadcast()


class BroadcastConsumer:
    def __init__(
        self,
        websocket: WebSocket,
        group_id: str,
        connection_type: str,
        topics_to_subscribe: List[str],
        consumer_poll: float = 10.0,
    ):
        self.websocket = websocket
        self.group_id = group_id
        self.connection_type = connection_type
        self.topics_to_subscribe = topics_to_subscribe

        self.poll = consumer_poll

    async def broadcast(self) -> None:
        if self.websocket.application_state == WebSocketState.DISCONNECTED:
            return
        await connection_manager.connect(self.websocket, self.connection_type)
        consumer = Consumer(
            group_id=self.group_id, topics_to_subscribe=self.topics_to_subscribe
        ).get_consumer()

        try:
            while True:
                message = consumer.poll(self.poll)
                if message is None:
                    continue
                if message.error():
                    logging.error("Consumer error: {}".format(message.error()))
                    continue

                await connection_manager.broadcast(message.value(), self.connection_type)
        except (WebSocketDisconnect, ConnectionClosedError, ConnectionClosedOK) as e:
            await self.websocket.close()
            connection_manager.disconnect(self.websocket, self.connection_type)


CONNECTION_TYPES = ["price"]

class ConnectionManager:
    """
    This manager will allow us to have several connections at the same time
    """

    def __init__(self) -> None:
        self.connections: dict[str: List[WebSocket]] = {conn: [] for conn in CONNECTION_TYPES}

    def get_connections(self, connection_type: str):
        return self.connections.get(connection_type)

    async def connect(self, websocket: WebSocket, connection_type: str) -> None:
        await websocket.accept()
        self.connections.get(connection_type).append(websocket)

    def disconnect(self, websocket: WebSocket, connection_type: str) -> None:
        self.connections.get(connection_type).remove(websocket)

    async def broadcast(self, data: str, connection_type: str) -> None:
        for connection in self.connections.get(connection_type):
            await connection.send_json(data)


connection_manager = ConnectionManager()
```

### Description

I
1.Open the postman(or other) websocket client
2.Connect to websocket(everything works as expected - messages are displayed in postman)
3.Disconnect websocket
4.Connect to the same websocket endpoint again
Expected: Messages are displayed in postman
Actual: Websocket doesn't respond
II
1.Open the two tabs in postman(or other) websocket client
2.Connect to websocket from first tab(everything works as expected - messages are displayed in postman)
3.Connect to websocket from second tab(Both of websocket connections stop working)
Expected: Messages are displayed in both postman tabs
Actual: Websocket doesn't respond

While debugging i have discovered that the messages are polled correctly and there seems to be a problem with websockets send_json() method but i am not sure how to proceed with this. Putting receive_json() call in the while loop seems to make things better(hangs still happen sometimes) and i don't think that it is proper way to work around that problem.

### Operating System

Other

### Operating System Details

docker image python:3.9-slim

### FastAPI Version

fastapi = "0.68.0"

### Python Version

python 3.9.4

**Comment by farlabjavier** [comment]
Some update ? facing the same in  "0.68.1"

**Comment by eryk-merix** [comment, marked as answer]
The problem wasn't actually caused by any problem with fastapi. Polling the message by the consumer wasn't async and it was hanging the connection and that caused all the issues with reconnect.
